pkg/controlplane/apiserver/config.go

```diff
@@ -213,18 +242,6 @@ func BuildAuthorizer(ctx context.Context, s controlplaneapiserver.CompletedOptio
 	return authorizer, ruleResolver, enablesRBAC, err
 }
 
-// CreatePeerEndpointLeaseReconciler creates a apiserver endpoint lease reconciliation loop
-// The peer endpoint leases are used to find network locations of apiservers for peer proxy
-func CreatePeerEndpointLeaseReconciler(c genericapiserver.Config, storageFactory serverstorage.StorageFactory) (reconcilers.PeerEndpointLeaseReconciler, error) {
-	ttl := controlplane.DefaultEndpointReconcilerTTL
-	config, err := storageFactory.NewConfig(api.Resource("apiServerPeerIPInfo"))
-	if err != nil {
-		return nil, fmt.Errorf("error creating storage factory config: %w", err)
-	}
-	reconciler, err := reconcilers.NewPeerEndpointLeaseReconciler(config, "/peerserverleases/", ttl)
-	return reconciler, err
-}
-
```

Comment:
note for reviewer: this is only invoked from `pkg/controlplane`. Hence, move it there for less dependencies.

Reply:
Note: this move will be reverted in the follow-up, moving this function and everything else related to peer proxying to `pkg/controlplane/apiserver`. This move here helps to make the followup more obvious.